### ISSUE TITLE
System information not appearing

### ISSUE THREAD

**Issue by paulmadore** [mike168m/Cysboard#26]
It will appear if I run the program as root. How can I fix this? 

![1492550935](https://cloud.githubusercontent.com/assets/6739462/25154350/ef6e5fbc-2455-11e7-9126-5ed923a1f36d.png)

**Comment by mike168m** [comment]
Hmm can you please tell me the distro you are running on. I have had similar problems with arch linux users.

**Comment by paulmadore** [comment]
I'm on Debian Sid:

$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 9.0 (stretch)
Release:	9.0
Codename:	stretch

**Comment by mike168m** [comment]
Do you have an /etc/lsb-release file on you system. If you do and you still have this problem, i might have to find another way to get that information for debian users.

**Comment by paulmadore** [comment]
It looks like I don't. I installed the "lsb-core" package and that didn't seem to help that "no modules" situation either. It's really weird that it will allow it to have the info for Root, but not otherwise. There's apparently something you can do with the sudoers file to whitelist a program, but I've never done it before and I'm nervous about editing that file willy nilly.

**Comment by mike168m** [comment]
Hmm, you can still access that file without running as root. Have you tired changing permissions on the file.
Try ` apt-get install lsb-release`

**Comment by paulmadore** [comment]
Already have lsb-release. Here's what my system shows:

![1492611062](https://cloud.githubusercontent.com/assets/6739462/25184466/271708bc-24e0-11e7-9047-a3bdcd0b8dfe.png)

Which file should I change perms on? Any of these? Because there doesn't appear to be anything in /etc.

**Comment by paulmadore** [comment]
I just want to say, also, that I think this program is WAY better than Conky and that styling with HTML/CSS is a brilliant move. I'd like to see that method expanded to a whole desklet system.

**Comment by mike168m** [comment]
Thanks.  It should be a file in the /etc directory. Thats where cysboard will look for it. If it does not find a file named `lsb-release` in /etc it will look for an `os-release` file in /etc. If it doesn't find either it will report an error. Can you please run cysboard from a terminal and post the output. That would help alot. I might have to add more logging to resolve future issues.

**Comment by paulmadore** [comment]
No errors:

![1492623582](https://cloud.githubusercontent.com/assets/6739462/25193606/49eeb3d6-24fd-11e7-9bda-ae28defe371e.png)

However, it appears that my os-release file is located in /usr/lib. 

Again, it shows the correct output when run as root:

![1492623646](https://cloud.githubusercontent.com/assets/6739462/25193653/704d5f28-24fd-11e7-8db6-a30db6e8b6ee.png)

Where is the line of code I can change to make it look in /usr/lib instead? Perhaps a good long-term fix for this is to have a configuration file associated with it that can change defaults like this.

**Comment by mike168m** [comment]
The code is in the `osinfo_linux.cpp` (constructor) and it needs some refactoring to search directories first. You can go ahead and create a pull request if you fix it. I will add a fix for this issue as soon as i can.
Thanks again for catching this.

**Comment by paulmadore** [comment]
Okay, I'll have a look. Glad to know it wasn't just bad configuration on my part.

**Comment by paulmadore** [comment]
Well, this is helpful information: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=444678

Debian does not have an /etc/lsb_release file. I'm wondering if I can create one myself.

**Comment by paulmadore** [comment]
By creating a file lsb-release with the following lines: 

DISTRIB_ID=Debian
DISTRIB_DESCRIPTION="LCC"
DISTRIB_RELEASE=9
DISTRIB_CODENAME=Stretch

I was able to get it to print Debian. This is quick hack until it is learned how to get the information in Debian. 

The ram still does not display without running as root. Looking into that next.

**Comment by paulmadore** [comment]
The ram issue was a problem with my template. 

What I am going to do is write a little script that generates an lsb-release file for Debian systems, and this can be an optional fix for Debian users whilst you create something more universal.

**Comment by mike168m** [comment]
Thanks for the work around. Looks like i'll have to have the files in a list generated at compile time for different distros and select from that at runtime.

**Comment by paulmadore** [comment]
Yeah, and that's a bit more work than I want to do at this point. I'm going to create a theme repository for Cysboard at lucky.computer/cysboard

For whatever reason Debian broke with the standardized way of providing system information. They minimally comply with the LSB standard by having output via the lsb_release command. My computer company will still be shipping Debian but I think I will have to do something about this issue, some script that generates a standard lsb-release file during updates. 

It's really great to see a conky alternative that is more fun to work with.
